shoop/admin/modules/orders/views/edit.py

```diff
@@ -332,6 +317,92 @@ def handle_finalize(self, request):
+        customer = Contact.objects.filter(pk=customer_id).first()
+        if not customer:
+            return JsonResponse(
+                {"success": False, "errorMessage": _("Contact %s does not exist.") % customer_id}, status=400
+            )
+        return {
+            "id": customer.id,
+            "name": customer.name,
+            "isCompany": bool(isinstance(customer, CompanyContact)),
+            "billingAddress": encode_address(customer.default_billing_address),
+            "shippingAddress": encode_address(customer.default_shipping_address)
+        }
+
+    def handle_customer_data(self, request):
+        customer_id = request.GET["id"]
+        return self.get_customer_data(customer_id)
+
+
+class PurchaseOrderEditView(BaseOrderEditView):
+    model = PurchaseOrder
+    order_type = "purchase"
+
+    def get_context_data(self, **kwargs):
+        context = super(PurchaseOrderEditView, self).get_context_data(**kwargs)
+        manufacturer = None
+        manufacturer_id = self.request.GET.get("manufacturer_id")
+        if manufacturer_id:
+            manufacturer = Manufacturer.objects.filter(pk=manufacturer_id).first()
+        context["config"].update({
+            "manufacturer": encode_manufacturer_data(manufacturer) if manufacturer else None
+        })
+        return context
+
+    def handle_manufacturer_data(self, request):
+        manufacturer_id = request.GET["id"]
+        manufacturer = Manufacturer.objects.filter(pk=manufacturer_id).first()
+        if not manufacturer:
+            return JsonResponse(
+                {"success": False, "errorMessage": _("Manufacturer %s does not exist.") % manufacturer_id}, status=400
+            )
+        return encode_manufacturer_data(manufacturer)
+
+    def handle_product_data(self, request):
+        product_data = super(PurchaseOrderEditView, self).handle_product_data(request)
+        product = Product.objects.filter(pk=product_data["id"]).first()
+        shop_id = request.GET["shop_id"]
+        shop = Shop.objects.get(pk=shop_id)
+        shop_product = product.get_shop_instance(shop)
+        supplier = shop_product.suppliers.first()  # TODO: Allow setting a supplier?
+        latest_adjustment = supplier.get_last_purchase_price(product.id)
+        purchase_price = (latest_adjustment.purchase_price if latest_adjustment else None)
+        if purchase_price:
+            # note - I'm storing purchase price in base unit price to avoid having to change the OrderLine
+            # model - hacky, I know...
+            # i think adding a purchase_price field to OrderLine could be useful not just for purchase orders
+            # but also for sales orders to eventually generate interesting cost analysis reports
+            # (ie income statement/profit margin report).
+            # Do you think its worth adding the field? Doing so will however mean I will need
+            # to extend SourceLines so that the correct prices are used
```

Comment:
I think you should use orderline base unit price to store the purchase price for this order. For sales order lines we can get the purchase price from supplier. And it's supplier job to calculate that based on stock adjustments and stock strategy (lifo, fifo, latest price etc...).